src/statistics/Statistics.ts

```diff
@@ -218,25 +295,79 @@ export class Statistics {
         }
     }
 
-    private addNearCacheStats(stats: string[]): void {
+    private addNearCacheStats(stats: string[], compressor: MetricsCompressor): void {
         for (const nearCache of this.client.getNearCacheManager().listAllNearCaches()) {
-            const nearCacheNameWithPrefix = this.getNameWithPrefix(nearCache.getName());
+            const name = nearCache.getName();
+            const nearCacheNameWithPrefix = this.getNameWithPrefix(name);
             nearCacheNameWithPrefix.push('.');
+            const nameWithPrefix = nearCacheNameWithPrefix.join('');
+
             const nearCacheStats = nearCache.getStatistics();
-            const prefix = nearCacheNameWithPrefix.join('');
-            this.addStat(stats, 'creationTime', nearCacheStats.creationTime, prefix);
-            this.addStat(stats, 'evictions', nearCacheStats.evictedCount, prefix);
-            this.addStat(stats, 'hits', nearCacheStats.hitCount, prefix);
-            this.addEmptyStat(stats, 'lastPersistenceDuration', prefix);
-            this.addEmptyStat(stats, 'lastPersistenceKeyCount', prefix);
-            this.addEmptyStat(stats, 'lastPersistenceTime', prefix);
-            this.addEmptyStat(stats, 'lastPersistenceWrittenBytes', prefix);
-            this.addStat(stats, 'misses', nearCacheStats.missCount, prefix);
-            this.addStat(stats, 'ownedEntryCount', nearCacheStats.entryCount, prefix);
-            this.addStat(stats, 'expirations', nearCacheStats.expiredCount, prefix);
-            this.addEmptyStat(stats, 'ownedEntryMemoryCost', prefix);
-            this.addEmptyStat(stats, 'lastPersistenceFailure', prefix);
+            this.addNearCacheMetric(stats, compressor,
+                'creationTime', name, nameWithPrefix, nearCacheStats.creationTime,
+                ValueType.LONG, ProbeUnit.MS);
+            this.addNearCacheMetric(stats, compressor,
+                'evictions', name, nameWithPrefix, nearCacheStats.evictedCount,
+                ValueType.LONG, ProbeUnit.COUNT);
+            this.addNearCacheMetric(stats, compressor,
+                'hits', name, nameWithPrefix, nearCacheStats.hitCount,
+                ValueType.LONG, ProbeUnit.COUNT);
+            this.addNearCacheMetric(stats, compressor,
+                'misses', name, nameWithPrefix, nearCacheStats.missCount,
+                ValueType.LONG, ProbeUnit.COUNT);
+            this.addNearCacheMetric(stats, compressor,
+                'ownedEntryCount', name, nameWithPrefix, nearCacheStats.entryCount,
+                ValueType.LONG, ProbeUnit.COUNT);
+            this.addNearCacheMetric(stats, compressor,
+                'expirations', name, nameWithPrefix, nearCacheStats.expiredCount,
+                ValueType.LONG, ProbeUnit.COUNT);
+
+            // kept for backwards-compatibility purposes with Management Center 4.0
+            this.addEmptyAttribute(stats, 'lastPersistenceDuration', nameWithPrefix);
```

Comment:
What happens when you don't send them? I see that in Python client, we don't send empty values for those below

Reply:
If we don't send them, MC won't receive them. But I didn't check how MC 4.0 behaves when they're not present. Going to do a quick check now.

Reply:
I tried to remove all null gauges and empty near cache stats and so no issues in MC 4.0 apart from https://github.com/hazelcast/management-center/issues/4165 which is a known issue.

So, I removed all of these gauges and got rid of `addEmptyAttribute()` method. See bf6d04783b65fef8d92a2fa8bb723a9408415ae9